databuilder/extractor/minio_stats_extractor.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Iterator, List, Set, Optional
+
+import boto3
+from pyhocon import ConfigTree
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.models.table_stats import TableColumnStats
+from databuilder.utils.minio_spec import MinioSpecUtils
+from databuilder.utils.pyspark_stats import get_stats_by_column
+
+
+class MinioStatsExtractor(Extractor):
+
+    """
+    An Extractor that extracts meta data from minio and stores them in amundsen.
+    """
+    # CONFIG KEYS
+    ACCESS_KEY = 'access_key'
+    SECRET_KEY = 'secret_key'
+    BUCKET_NAME = 'bucket_name'
+    ENDPOINT_URL = 'endpoint_url'
+    SPARK_SESSION_KEY = 'spark_session'
+
+    def init(self, conf: ConfigTree) -> None:
+        """
+        :param conf:
+        """
+        self.conf = conf
+        self._extract_iter: Optional[Iterator[TableColumnStats]] = None
+
+        self.bucket_name = conf.get_string(MinioStatsExtractor.BUCKET_NAME)
+        self.endpoint_url = conf.get_string(MinioStatsExtractor.ENDPOINT_URL)
+        self.access_key = conf.get_string(MinioStatsExtractor.ACCESS_KEY)
+        self.secret_key = conf.get_string(MinioStatsExtractor.SECRET_KEY)
+        self.spark_session = conf.get(MinioStatsExtractor.SPARK_SESSION_KEY)
+
+        self.s3 = boto3.resource('s3',
+                                 endpoint_url=self.endpoint_url,
+                                 aws_access_key_id=self.access_key,
+                                 aws_secret_access_key=self.secret_key,
+                                 region_name='us-east-1')
```

Comment:
Would be nice to deduplicate the config points / boto3 resource creation parts of this from the minio_extractor